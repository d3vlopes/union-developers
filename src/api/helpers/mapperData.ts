import { MainLayoutProps } from '@/presentation/layouts'

import { MainQuery } from '@/api/generated/graphql'

import * as mappers from '@/presentation/mappers'

export function mapperData(remoteData: MainQuery): MainLayoutProps {
  const api = remoteData.pages[0]

  const data: MainLayoutProps = {
    base: {
      header: mappers.HeaderMapper.toDomain(api.header),
      footer: mappers.FooterMapper.toDomain(api.footer),
    },
    hero: mappers.HeroMapper.toDomain(api.hero),
    aboutSection: mappers.AboutSectionMapper.toDomain(
      api.aboutSection,
    ),
    pillarsSection: mappers.PillarsSectionMapper.toDomain(
      api.pillarsSection,
    ),
    highlightSection: mappers.HighlightSectionMapper.toDomain(
      api.benefitsSection,
    ),
    stepsSection: mappers.StepsSectionMapper.toDomain(
      api.stepsSection,
    ),
    rememberActionSection:
      mappers.RememberActionSectionMapper.toDomain(
        api.rememberActionSection,
      ),
    testimonialSection: mappers.TestimonialSectionMapper.toDomain(
      api.testimonialSection,
    ),
    faqSection: mappers.FaqSectionMapper.toDomain(api.faqSection),
  }

  return data
}
