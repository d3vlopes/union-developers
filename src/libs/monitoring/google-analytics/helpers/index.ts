type AnalityClickProps =
  | Gtag.CustomParams
  | Gtag.ControlParams
  | Gtag.EventParams
  | undefined

export function handleAnalityClick(eventParams: AnalityClickProps) {
  gtag('event', 'click', eventParams)
}

export function handleAnalitySubscriptions(
  eventParams: AnalityClickProps,
) {
  gtag('event', 'subscription', eventParams)
}
