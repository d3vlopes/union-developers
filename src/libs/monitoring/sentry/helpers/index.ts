import { Sentry } from '..'

interface CaptureMessageArgs extends Sentry.Breadcrumb {
  user?: Sentry.User
  error?: unknown
  logMessage?: string
}

export function captureMessage({
  category,
  message,
  level,
  user,
  error,
  logMessage,
}: CaptureMessageArgs) {
  Sentry.addBreadcrumb({
    category,
    message,
    level,
  })

  if (user) Sentry.setUser(user)
  if (logMessage) Sentry.captureMessage(logMessage)
  if (error) {
    Sentry.setUser(user || null)
    Sentry.captureException(error)
  }
}
