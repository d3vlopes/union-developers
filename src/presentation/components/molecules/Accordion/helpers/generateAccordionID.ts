export function generateAccordionID(value: string) {
  return value
    .trim()
    .toLowerCase()
    .replaceAll(' ', '-')
    .replace('?', '')
    .normalize('NFD')
    .replace(/[\u0300-\u036f]/g, '')
}
