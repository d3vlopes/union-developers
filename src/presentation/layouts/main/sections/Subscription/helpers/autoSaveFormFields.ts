import { Dispatch, SetStateAction } from 'react'

export function autoSaveFormFields<TFormValues, TFormData>(
  formValues: TFormValues,
  formData: TFormData,
  errors: object,
  setFormData: Dispatch<SetStateAction<TFormData>>,
  delay = 5000,
) {
  const interval = setInterval(() => {
    const isFieldContainError = Object.keys(errors).length > 0

    if (!isFieldContainError) {
      setFormData({ ...formData, ...formValues })
    }
  }, delay)

  return interval
}
