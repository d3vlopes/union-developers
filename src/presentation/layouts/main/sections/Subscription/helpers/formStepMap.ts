export const formStepMap: Record<number, string> = {
  0: 'Inicial',
  1: 'Dados pessoais',
  2: 'Informações técnicas',
  3: 'Sobre você',
}
