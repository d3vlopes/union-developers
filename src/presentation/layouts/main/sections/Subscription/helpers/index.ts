export * from './autoSaveFormFields'
export * from './formStepMap'
