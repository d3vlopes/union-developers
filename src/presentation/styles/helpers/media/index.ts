/* eslint-disable @typescript-eslint/no-explicit-any */
import { css } from 'styled-components'

import { Media, Breakpoints } from './types'

import { sizes } from './sizes'

export const media: Media = Object.keys(sizes).reduce<Media>(
  (acc, label) => {
    acc[label as Breakpoints] = (
      literals: TemplateStringsArray,
      ...args: unknown[]
    ) => css`
      @media (min-width: ${sizes[label as Breakpoints]}px) {
        ${css(literals, ...(args as any))};
      }
    `
    return acc
  },
  {} as Media,
)
