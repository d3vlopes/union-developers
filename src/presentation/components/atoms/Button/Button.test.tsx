import { render, screen } from '@/__tests__/helpers'

import { Button } from '.'

const { getByRole, getByText } = screen

describe('<Button />', () => {
  it('should render primary variant by default', () => {
    render(<Button>Children</Button>)

    const button = getByRole('button')
    const text = getByText('Children')

    expect(text).toBeInTheDocument()
    expect(button).toHaveStyleRule(
      'background',
      'linear-gradient(225deg,#0048FE 0%,#851D86 100%)',
      {
        modifier: '::before',
      },
    )
  })

  it('should render secondary variant', () => {
    render(<Button variant="secondary">Children</Button>)

    const button = getByRole('button')
    const text = getByText('Children')

    expect(text).toBeInTheDocument()
    expect(button).toHaveStyle({
      background: 'none',
      'border-right': '2px solid #0048FE',
      'border-left': '2px solid #851D86',
      'background-image':
        'linear-gradient(to right, #851D86, #0048FE),linear-gradient(to right, #851D86, #0048FE)',
    })
  })

  it('should render disabled', () => {
    render(<Button disabled>Children</Button>)

    const button = getByRole('button')
    const text = getByText('Children')

    expect(text).toBeInTheDocument()

    expect(button).toHaveStyle({
      opacity: 0.7,
      cursor: 'not-allowed',
    })
  })

  it('should match snapshot', () => {
    const { container } = render(<Button>Children</Button>)

    expect(container).toMatchSnapshot()
  })
})
