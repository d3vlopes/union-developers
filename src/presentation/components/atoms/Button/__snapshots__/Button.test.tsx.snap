// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should match snapshot 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  border-radius: 5px;
  width: 100%;
  background: none;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
  box-shadow: 0px 4px 4px 0px rgba(0,0,0,0.25);
  overflow: hidden;
  height: auto;
}

.c1 span {
  color: #FFFFFF;
  font-size: 2.2rem;
  font-family: Poppins,apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Open Sans","Helvetica Neue",sans-serif;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  font-weight: 700;
  z-index: 2;
}

.c1:disabled {
  opacity: 0.7;
  cursor: not-allowed;
}

.c1::before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: 5px;
  background: linear-gradient(225deg,#0048FE 0%,#851D86 100%);
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  z-index: 1;
}

@media (min-width:1024px) {
  .c1:hover:not(.c0:disabled) {
    border-right: 2px solid #0048FE;
    border-left: 2px solid #851D86;
    background-image: linear-gradient(to right,#851D86,#0048FE), linear-gradient(to right,#851D86,#0048FE);
    background-size: 100% 2px;
    background-position: 0 100%,0 0;
    background-repeat: no-repeat;
    background-clip: padding-box;
  }

  .c1:hover:not(.c0:disabled):hover {
    -webkit-filter: brightness(0.9);
    filter: brightness(0.9);
  }

  .c1:hover:not(.c0:disabled)::before {
    left: 100%;
  }
}

<div>
  <button
    class="c0 c1"
  >
    <span>
      Children
    </span>
  </button>
</div>
`;
